Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AdAuthManager.java

```
@@ -76,7 +80,15 @@ public String getAccessToken(String tid) throws IOException {
      */
     public String getAccessToken(String tid, String resource, PromptBehavior promptBehavior) throws IOException {
         AuthContext ac = createContext(tid, null);
-        AuthResult result = ac.acquireToken(resource, false, adAuthDetails.getAccountEmail(), false);
+        AuthResult result = null;
+        try {
+            result = ac.acquireToken(resource, false, adAuthDetails.getAccountEmail(), false);
+        } catch (AuthException e) {
+            if (AuthError.InvalidGrant.equalsIgnoreCase(e.getError())
```

we can throw the exception if not going into the if block.